packages/react-native-dogfood/src/components/MeetingUI.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import { NativeStackScreenProps } from '@react-navigation/native-stack';
+import {
+  ActiveCall,
+  ActiveCallProps,
+  useCall,
+} from '@stream-io/video-react-native-sdk';
+import { MeetingStackParamList, ScreenTypes } from '../../types';
+import { Button, SafeAreaView, StyleSheet, Text, View } from 'react-native';
+import { theme } from '@stream-io/video-react-native-sdk/dist/src/theme';
+import { ParticipantListButtons } from '../components/ParticipantListButtons';
+import { LobbyViewComponent } from './LobbyViewComponent';
+import { AuthenticationProgress } from './AuthenticatingProgress';
+
+type Props = NativeStackScreenProps<
+  MeetingStackParamList,
+  'MeetingScreen' | 'GuestMeetingScreen'
+> & {
+  callId: string;
+  show: ScreenTypes;
+  setShow: React.Dispatch<React.SetStateAction<ScreenTypes>>;
+};
+type Mode = NonNullable<ActiveCallProps['mode']>;
+
+export const MeetingUI = ({
+  callId,
+  navigation,
+  route,
+  show,
+  setShow,
+}: Props) => {
+  const [selectedMode, setSelectedMode] = React.useState<Mode>('grid');
+  const call = useCall();
+
+  const returnToHomeHandler = () => {
+    navigation.navigate('JoinMeetingScreen');
+  };
+
+  const backToLobbyHandler = () => {
+    setShow('lobby');
+  };
+
+  let ComponentToRender: JSX.Element | null = null;
```

Comment:
instead of let I think it is easier to  use early return

of course optional for you to change. 

Example:

```
else if (show === 'lobby') {
   return (
      <LobbyViewComponent callId={callId} {...{ navigation, route }} />
    );
```